packages/mcl/src/main.d

```diff
@@ -1,73 +1,89 @@
+module mcl.main;
+
 import std.stdio : writefln, writeln, stderr;
 import std.array : replace;
 import std.getopt : getopt;
 import std.logger : infof, errorf, LogLevel;
-
+import std.sumtype : SumType, match;
+import std.string : stripRight, stripLeft;
+import std.algorithm : endsWith;
+import std.format : format;
 import mcl.utils.path : rootDir;
 import mcl.utils.tui : bold;
 
-import cmds = mcl.commands;
+import mcl.commands;
 
-alias supportedCommands = imported!`std.traits`.AliasSeq!(
-    cmds.get_fstab,
-    cmds.deploy_spec,
-    cmds.ci_matrix,
-    cmds.print_table,
-    cmds.shard_matrix,
-    cmds.host_info,
-    cmds.ci,
-    cmds.machine,
-    cmds.config,
-);
+import argparse;
 
-int main(string[] args)
-{
-    if (args.length < 2)
-        return wrongUsage("no command selected");
 
-    string cmd = args[1];
-    LogLevel logLevel = LogLevel.info;
-    args.getopt("log-level", &logLevel);
+@(Command(" ").Description(" "))
+struct unknown_command_args {}
```

Comment:
Prefer capitalized names ([`PascalCase`](https://dlang.org/dstyle.html#naming_classes), to be exact) for `struct`s. Applies to the rest of the `_args` `struct`s too.